File: python/nt_solutions/segmentation_3_snakes_levelset/exo6.py
plt.figure(figsize=(10,10))
phi = np.copy(phi0)
k = 0
gW = grad(W, order=2)

for i in range(1,niter+1):
    gD = grad(phi, order=2)
    d = np.maximum(eps*np.ones([n,n]), np.sqrt(np.sum(gD**2, 2)))
    g = gD/np.repeat(d[:,:,np.newaxis], 2, 2)
    G = W*d*div(g[:,:,0], g[:,:,1], order=2) + np.sum(gW*gD,2)
    phi = phi + tau*G
    if i % 30 == 0:
        phi = perform_redistancing(phi)
    if i % int(niter/4.) == 0:
        k = k + 1
        plt.subplot(2, 2, k)
        plot_levelset(phi,0,f0)

File: r/nt_solutions/segmentation_3_snakes_levelset/exo6.R




par(mfrow=c(2,2))


phi <- phi0
k <- 1
gW <- grad(W, order=2)

for (i in 1:niter){
  gD <- grad(phi, order=2)
  d <- pmax(eps*array(1, c(n,n)), sqrt(apply(gD**2, c(1,2), sum)))
  g <- gD/array(rep(d,2), c(dim(d),2))
  G <- W*d*div(g[,,1], g[,,2], order=2) + apply(gW*gD, c(1,2), sum)
  phi <- phi + tau*G
  if (mod(i, as.integer(niter/4))==0){
    k <- k+1
    plot_levelset(phi, f0, lw=2)
  }
  
}

